### ISSUE TITLE
Pipe.scan is not single-threaded...

### ISSUE THREAD

**Issue by haf** [logary/logary#310]
...as such it cannot reliably be used for counting/buffering, etc.

Try outputting two events and using the scan implementation I posted in an earlier issue; the initial state is created twice. Experimentally, 8 different state spaces are created, which seems to correspond to the # of Hopac threads.

**Comment by haf** [comment]
Ref https://github.com/logary/logary/issues/310 — this needs to be fixed to have serialisability across loggers' shipped Message values.

**Comment by lust4life** [comment]
pipe `state` will be created when the the pipe had been build, which will only occur in `registry` init time. Perhaps this is due to the current implementation about the `Pipe.compose` . which chains all pipes not dispatch them. #378

**Comment by haf** [comment]
Yes, it's probably because each pipe segment is run on Hopac and Hopac chooses where to run it.
